docs/pkgtorrent-service.8

```diff
@@ -0,0 +1,53 @@
+.TH pkgtorrent-service 8 "09 MARCH 2016" ""
+.SH NAME
+.B pkgtorrent-service(8)
+Is a python WSGI web application that can generate BitTorrent files for binary files available from the web server.
+.SH SYNOPSIS
+The program should installed on a site local
+.B yum
+mirror and must be configured before use. See \fBpkgtorrent-service.conf(5)\fR for more information.
+.SH USAGE
+The primary use case for this service is implemented in \fByum-torrent(1)\fR.
+.PP
+There is a secondary interface to assist with simple file downloads involving a single path.
+.SH EXAMPLE
+.RS
+.nf
+$ curl http://example.com/\fBpath/to/large/file\fR > \fBfile\fR
+.fi
+.RE
+.PP
+can be converted to using BitTorrent with:
+.PP
+.RS
+.nf
+$ curl http://example.com/\fBpath\fR/torrent_service/single/\fBpath/to/large/file\fR > \fBfile\fR.torrent
+$ btclient \fBfile\fR.torrent
+.fi
+.RE
+.PP
+Note the repetition of \fBpath\fR before the service name, and the addition of \fBtorrent_service/single\fR to pass control to the single file mode.
```

Comment:
I agree this is not worded well.

The reason for repeating the path is to permit single *path* mode to still return a multi-file torrent with appropriate nullfile/padding synthesized files. All of these must lie within the same url prefix (from url-list) and the torrent "name" which is `path` on multi-file torrents.

Reply:
Perhaps I'm missing something, but I couldn't find any piece of code where the actual `path` prefix to `torrent_service` in the URL would have any effect, other than the mentioned `nulldir` path (which is not used to write anything anyway; it's only read).

The prefix that is used as a torrent name for multi-file torrents is taken from the `p` field bundled in the POST request that the client sends.

This is what the server does:

1. get `p` from POST (via `form` constructed from `environ`) and store it in `pathspecs`
2. generate the actual `paths` list using `pathspecs` (expanding any globs etc.)
3. generate the `prefixes` list from `paths`
4. use the first element from `prefixes` as the `name` attribute in the newly created torrent file

The path prefix in the URL is only available in two `environ` fields: `SCRIPT_NAME` and `REQUEST_URI`. We only read the former and we strip any prefix so that only `torrent_service` remains.

Again, I might as well be missing something, so please correct me if so :)

Reply:
Also, in single mode, `pathspecs` is initialized from `path_info` (which starts with `/single`) by stripping the `/single` part.